skada/_utils.py

```diff
@@ -114,13 +182,64 @@ def check_X_domain(
     return_indices: bool = False,
     # xxx(okachaiev): most likely this needs to be removed as it doesn't fit new API
     return_joint: bool = True,
-    allow_auto_sample_domain: bool = False,
+    allow_auto_sample_domain: bool = True,
 ):
+    """
+    Input validation for domain adaptation (DA) estimator.
+    If we work in single-source and single target mode, return source and target
+    separately to avoid additional scan for 'sample_domain' array.
+
+    Parameters:
+    ----------
+    X : array-like of shape (n_samples, n_features)
+        Input features.
+    sample_domain : array-like of shape (n_samples,)
+        Domain labels for each sample.
+    allow_domains : set of int, optional (default=None)
+        Set of allowed domain labels. If provided, only these domain labels are allowed.
+    allow_source : bool, optional (default=True)
+        Allow the presence of source domains.
+    allow_multi_source : bool, optional (default=True)
+        Allow multiple source domains.
+    allow_target : bool, optional (default=True)
+        Allow the presence of target domains.
+    allow_multi_target : bool, optional (default=True)
+        Allow multiple target domains.
+    return_indices : bool, optional (default=False)
+        If True, return only the source indices.
+    return_joint : bool, optional (default=True)
+        If True, return X and sample_domain without separation.
+    allow_auto_sample_domain : bool, optional (default=True)
+        Allow automatic generation of sample_domain if not provided.
+
+    Returns:
+    ----------
+    If return_indices is True:
```

Comment:
do we erally want the output of the function to change this much? The way I see that this function does 4 different thgigs:
- input validation (is everythin OK), this is always done and raise error if not
- Return info or data
  - indices
  - X_source/X_taregt
  - Return joint (with pitentially updated sample_domain if None)
  
 I think we should factorized the data validztion and have 3 functiosn for the 3 different return values maybe? We can talk about this i slack